### ISSUE TITLE
Emacs hangs opening org file

### ISSUE THREAD

**Issue by bobf32** [doomemacs/doomemacs#7899]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.
- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

When I open an org file, it should display in the GUI window with appropriate font faces

### Current behavior

When I open a specific org file, emacs hangs on opening 

### Steps to reproduce

1. Open emacs
2. ```SPC-f-r``` to open recent file
3. Select the file in question
4. Hit Enter
5. Emacs hangs with the recent file selection window visible
6. Wait for about ten seconds
7. Hit ```C-g```. The following message appears: ```'org-element--parse-to': Suppressed 'C-g'. Press 'C-g' four more times to force interruption.```
8. Press ```C-g``` four more times
9. ```SPC-f-r``` to open recent file
10. Select the file in question
11. Hit Enter
12. File loads, but with no syntax highlighting

### System Information

https://pastebin.com/TnLB5z4D

**Comment by seokbeomKim** [comment]
I guess you have a wrong configuration. I also got the same issue when I have configured 'wrong font' in the configuration file. Please check the font is installed in your system (you can check through 'fc-list | grep -i <font name>'.

**Comment by hlissner** [comment]
I am unable to reproduce this. Are you still experiencing this issue? If you are:

1. [Check \*Messages\* for any other errors or warnings](https://discourse.doomemacs.org/t/how-to-debug-issues/55) that might've been emitted besides `'org-element--parse-to': Suppressed 'C-g'. Press 'C-g' four more times to force interruption.`
2. Try [using the profiler](https://discourse.doomemacs.org/t/how-to-debug-issues/55) (activate it before loading the org file, then again after you've aborted it). This should hopefully tell you where it spent most of its time during that period.
3. Try upgrading to at least Emacs 28.x, just in case this is an upstream issue (not to mention, Doom will be dropping 27.x support in the coming months).

**Comment by ning-y** [comment]
**tl;dr try `(package! org :pin "8ef6205")` in `packages.el`.**

I encountered a similar issue generating my org-agenda. Using `toggle-debug-on-quit` I found that the function reliable for the hanging is `org-element-cache-map`. Through a search I found a similar [bug description and diff](https://mail.gnu.org/archive/html/emacs-elpa-diffs/2022-09/msg00125.html) submitted to ELPA for this function: apparently, GC is disabled during some part of `org-element-cache-map` and this causes "Emacs [to] hang in some particularly bad scenarios."

Unfortunately it seems the diff was never merged. However, manually redefining `org-element-cache-map` to disable the GC therefore matching the submitted diff fixed the issue for me. As a more permanent fix I've pinned org to the last release before GC was disabled in `org-element-cache-map` (release 9.5.5 == commit 8ef6205).

**Comment by khorser** [comment]
I'm experiencing the same issue. Emacs 29.4, Windows 11 with 64GiB of RAM. Attached is a sample file for `org-passwords` which almost always causes the loop in `org-element--parse-to` on decryption and/or encryption when opened with `SPC u SPC u org-passwords`

[passwords_gpg.txt](https://github.com/user-attachments/files/17900744/passwords_gpg.txt)

@ning-y 
Thanks a lot for sharing, you saved my day (more than one day actually). As another workaround this helped me (not tested on huge org files):

```emacs-lisp
(defun my/restore-gc (fn &rest args)
  (let ((gc-cons-threshold #x1000000)) (apply fn args)))
(advice-add #'org-element--parse-to :around #'my/restore-gc)
```

**Comment by hlissner** [comment]
Not sure when it was fixed, but `org-element-cache-map` now [suppresses the GC](https://github.com/emacs-straight/org-mode/blob/main/lisp/org-element.el#L8031) so I'll consider this resolved.
